phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java

```diff
@@ -801,40 +747,155 @@ public int run(String[] args) throws Exception {
+        boolean verify = cmdLine.hasOption(VERIFY_OPTION.getOpt());
+
+        if (useTenantId) {
+            tenantId = cmdLine.getOptionValue(TENANT_ID_OPTION.getOpt());
+        }
+        if(useStartTime) {
+            startTime = new Long(cmdLine.getOptionValue(START_TIME_OPTION.getOpt()));
+        }
+        if (useEndTime) {
+            endTime = new Long(cmdLine.getOptionValue(END_TIME_OPTION.getOpt()));
+        }
+        if(isTimeRangeSet(startTime, endTime)) {
+            validateTimeRange();
+        }
+        if (verify) {
+            String value = cmdLine.getOptionValue(VERIFY_OPTION.getOpt());
+            indexVerifyType = IndexVerifyType.fromValue(value);
+        }
+        schemaName = cmdLine.getOptionValue(SCHEMA_NAME_OPTION.getOpt());
+        dataTable = cmdLine.getOptionValue(DATA_TABLE_OPTION.getOpt());
+        indexTable = cmdLine.getOptionValue(INDEX_TABLE_OPTION.getOpt());
+        isPartialBuild = cmdLine.hasOption(PARTIAL_REBUILD_OPTION.getOpt());
+        qDataTable = SchemaUtil.getQualifiedTableName(schemaName, dataTable);
+        basePath = cmdLine.getOptionValue(OUTPUT_PATH_OPTION.getOpt());
+        isForeground = cmdLine.hasOption(RUN_FOREGROUND_OPTION.getOpt());
+        useSnapshot = cmdLine.hasOption(SNAPSHOT_OPTION.getOpt());
+        shouldDeleteBeforeRebuild = cmdLine.hasOption(DELETE_ALL_AND_REBUILD_OPTION.getOpt());
+    }
+
+    private void validateTimeRange() {
+        Long currentTime = EnvironmentEdgeManager.currentTimeMillis();
+        Long st = (startTime == null) ? 0 : startTime;
+        Long et = (endTime == null) ? currentTime : endTime;
+        if (st.compareTo(currentTime) > 0 || et.compareTo(currentTime) > 0 || st.compareTo(et) >= 0) {
+            throw new RuntimeException(INVALID_TIME_RANGE_EXCEPTION_MESSAGE);
+        }
+    }
+
+    private Connection getConnection(Configuration configuration) throws SQLException {
+        return ConnectionUtil.getInputConnection(configuration);
+    }
+
+    private void setupIndexAndDataTable(Connection connection) throws SQLException {
+        pDataTable = PhoenixRuntime.getTableNoCache(connection, qDataTable);
+        if (!isValidIndexTable(connection, qDataTable, indexTable, tenantId)) {
+            throw new IllegalArgumentException(
+                    String.format(" %s is not an index table for %s for this connection",
+                            indexTable, qDataTable));
+        }
+        pIndexTable = PhoenixRuntime.getTable(connection, schemaName != null && !schemaName.isEmpty()
+                ? SchemaUtil.getQualifiedTableName(schemaName, indexTable) : indexTable);
+        indexType = pIndexTable.getIndexType();
+        if (schemaName != null && !schemaName.isEmpty()) {
+            qIndexTable = SchemaUtil.getQualifiedTableName(schemaName, indexTable);
+        } else {
+            qIndexTable = indexTable;
+        }
+        if (IndexType.LOCAL.equals(indexType)) {
+            isLocalIndexBuild = true;
```

Comment:
nit: I recommend adding this. for class level variables

Reply:
This is prevalent in the file. Let's get to this in the next change in the file.